client/cached_provider.go

```diff
@@ -150,17 +170,27 @@ func (cp *cachedProvider) GetJSON(ctx context.Context, key string, namespace str
 	if err := json.Unmarshal(bytes, result); err != nil {
 		return errors.Wrapf(err, "failed to unmarshal json into go type %T", result)
 	}
+	// json values are not logged
+	cp.otel.addTracingEvent(ctx, key, "", cfg.CommitSHA)
 	return nil
 }
 
 func (cp *cachedProvider) GetProto(ctx context.Context, key string, namespace string, result protoreflect.ProtoMessage) error {
-	return cp.store.Evaluate(key, namespace, fromContext(ctx), result)
+	cfg, err := cp.store.Evaluate(key, namespace, fromContext(ctx), result)
+	if err != nil {
+		return err
+	}
+	// proto values are not logged
+	cp.otel.addTracingEvent(ctx, key, "", cfg.CommitSHA)
+	return nil
 }
 
 func (cp *cachedProvider) GetString(ctx context.Context, key string, namespace string) (string, error) {
 	dest := &wrapperspb.StringValue{}
-	if err := cp.store.Evaluate(key, namespace, fromContext(ctx), dest); err != nil {
+	cfg, err := cp.store.Evaluate(key, namespace, fromContext(ctx), dest)
+	if err != nil {
 		return "", err
 	}
+	cp.otel.addTracingEvent(ctx, key, dest.GetValue(), cfg.CommitSHA)
```

Comment:
one can argue that logging strings could be dangerous, but let's worry about it later (we can have a flag for this)